Vitess user survey blog (#1832)
vitessio/website: content/en/blog/2024-08-27-vitess-user-survey.md

---
author: 'Kirtan Chandak'
date: 2024-08-27
slug: '2024-08-27-vitess-user-survey'

> tags: ['Vitess', 'PlanetScale', 'MySQL', 'VReplication', 'CDC', 'Debezium']

Reviewer: ```suggestion
tags: ['Vitess', 'Sharding', 'MySQL']
```

title: 'Vitess User Survey'

> description: "Learn the basics of Change Data Capture (CDC) and how to leverage Vitess's VStream API to build data pipelines."

Reviewer: Can you rewrite this? Looks like copy/paste.

---

We recently conducted a survey of how Vitess is being used by the community. This blog post summarizes what we learned.

> # What our users said - 

Reviewer: Let's remove this heading.

> **"Vitess solves an existential threat for services which outgrow a single MySQL database."**

> **"Horizontal sharding helps us scale quickly, and the new generation execution plan, Gen4, enables us to support more SQL queries."**

# Background

The story begins with the Vitess maintainer team wanting to conduct a user survey with community members and engineers using Vitess in production to gain a better understanding of how to improve the project. As an LFX Mentee'24, I worked with the Vitess maintainers to form the questions and create a format for conducting the user survey. We drafted the questions, reviewed them, and developed a final version of the survey. I'm happy to report that we received a number of responses.

> # Summary Responses  - 

Reviewer: ```suggestion
# Summary Responses
```

### 12. Have you run a resharding process?
A majority of the Vitess users have gone through a resharding process. 
<div style="width:500px;">
<img src="/files/2024-08-27-vitess-user-survey/12-resharding.png" alt="Vitess Project Booth" style="margin-top:6px"/>
</div>

### 13. Do you contribute code to Vitess?
50% of our respondents contribute code to Vitess. 
<div style="width:500px;">
<img src="/files/2024-08-27-vitess-user-survey/13-contribute.png" alt="Vitess Project Booth" style="margin-top:6px"/>
</div>

### 14. Cloud
50% of our users use Public cloud and most of them use GCP and Azure.
<div style="width:500px;">
<img src="/files/2024-08-27-vitess-user-survey/14-cloud.png" alt="Vitess Project Booth" style="margin-top:6px"/>
</div>

### 15. Which major version of Vitess are you running? 
23% of the users are running version 19.0 and other versions has a distributed split as seen in the graph below.
<div style="width:500px;">
<img src="/files/2024-08-27-vitess-user-survey/15-version.png" alt="Vitess Project Booth" style="margin-top:6px"/>
</div>

# Insights

### We asked our users how Vitess has helped their business. We got some interesting answers from different companies.

> 1. Scalability: Generally, it has scaled the performance of databases by horizontal sharding.

Reviewer: ```suggestion
1. Scalability: Many Vitess users have scaled the performance of their databases using horizontal sharding.
```

> For eg. -

Reviewer: ```suggestion
For example:
```

### We also asked them what the biggest missing feature in Vitess is. Here is a curated sample of the responses. 

1. Improved Observability: A consistent dashboard for observing the vreplication streams, with controls over schema changes, can give a better overview of operations.
2. Unified Web UI: A single, easy-to-use web interface for all administrative tasks would ease management and enhance usability.
3. Circuit Breaking and Adaptive Concurrency Limits: We need features to prevent clients from being blocked by slow shards, and manage concurrency.
4. More Clear Upgrade Path with LTS Versions: a well-defined upgrade path with LTS versions is important for stability and fewer bugs.
5. Auto Sharding Improvements: More advanced auto-sharding capabilities would be nice, such as in TiDB.
6. Vitess Keyspace History: Add a feature to track and review changes over time to a keyspace—such as reparents, backups, or scaling events for historical analysis and debugging.

# Conclusion

> We appreciate all of you for your time to fill up the survey and share the inputs from your side. Your feedback has provided us with crucial findings that will guide our future efforts.

Reviewer: ```suggestion
The maintainer team appreciates your responses to the survey. Your feedback will be incorporated into the project roadmap.
```

Author: made the changes.

---
author: 'Kirtan Chandak, Deepthi Sigireddi'
date: 2024-08-27
slug: '2024-08-27-vitess-user-survey'
tags: ['Vitess', 'PlanetScale', 'MySQL', 'VReplication', 'CDC', 'Debezium']
title: 'Vitess User Survey'
description: "Learn the basics of Change Data Capture (CDC) and how to leverage Vitess's VStream API to build data pipelines."
---

We recently conducted a survey of how Vitess is being used by the community. This blog post summarizes what we learned.

## Background

The story begins with the Vitess maintainer team wanting to conduct a user survey with community members and engineers using Vitess in production to gain a better understanding of how to improve the project. As an LFX Mentee'24, I worked with the Vitess maintainers to form the questions and create a format for conducting the user survey. We drafted the questions, reviewed them, and developed a final version of the survey. I'm happy to report that we received a number of responses.

> ## Summary Responses  - 

Reviewer: ```suggestion
## Summary Responses
```

> #### 1.How long have you been using Vitess?

Reviewer: ```suggestion
#### How long have you been using Vitess?
```

First of all, we decided to ask how long our community members have been using Vitess. Interestingly most users have been using Vitess for more than 3 years. 

> ![1-vitess-long.png](/files/2024-08-27-vitess-user-survey/1-vitess-long.png)

Reviewer: Can we center this instead of it being left justified?

Author: added different charts and it was not looking good in the middle so I kept it left-aligned, you can check the preview, and then make a decision
